presto-clickhouse/src/test/java/com/facebook/presto/plugin/clickhouse/TestClickhouseIntegrationMixedCase.java

```diff
@@ -0,0 +1,173 @@
+        assertFalse(getQueryRunner().tableExists(session, "Test_CreateAs_Mixed_Join"));
+    }
+
+    @Test
+    public void testDuplicatedColumNameCreateTable()
+    {
+        getQueryRunner().execute(session, "CREATE TABLE test (a integer, A integer)");
+        assertTrue(getQueryRunner().tableExists(session, "test"));
+
+        getQueryRunner().execute(session, "CREATE TABLE TEST (a integer, A integer)");
+        assertTrue(getQueryRunner().tableExists(session, "TEST"));
+
+        assertQueryFails("CREATE TABLE Test (a integer, a integer)", "line 1:31: Column name 'a' specified more than once");
+        assertFalse(getQueryRunner().tableExists(session, "Test"));
+
+        assertUpdate(session, "DROP TABLE IF EXISTS test");
+        assertFalse(getQueryRunner().tableExists(session, "test"));
+
+        assertUpdate(session, "DROP TABLE IF EXISTS TEST");
+        assertFalse(getQueryRunner().tableExists(session, "TEST"));
+    }
+
+    @Test
+    public void testSelect()
+    {
+        getQueryRunner().execute(session, "CREATE TABLE Test_Select  AS SELECT * FROM tpch.region where regionkey=3");
+        assertTrue(getQueryRunner().tableExists(session, "Test_Select"));
+        assertQuery("SELECT * from default.Test_Select", "VALUES (3, 'EUROPE', 'ly final courts cajole furiously final excuse')");
+
+        getQueryRunner().execute(session, "CREATE TABLE test_select  AS SELECT * FROM tpch.region LIMIT 2");
+        assertQuery("SELECT COUNT(*) FROM default.test_select", "VALUES 2");
+
+        getQueryRunner().execute(session, "DROP TABLE IF EXISTS TEST_SELECT");
+        getQueryRunner().execute(session, "DROP TABLE IF EXISTS Test_Select");
+    }
+
+    @Test
+    public void testAlterTable()
+    {
+        assertUpdate(" CREATE TABLE Test_Alter (x int NOT NULL, y int, a int) WITH (engine = 'MergeTree', order_by = ARRAY['x'])");
+        assertTrue(getQueryRunner().tableExists(getSession(), "Test_Alter"));
+
+        assertQueryFails("ALTER TABLE Test_Alter RENAME COLUMN Y to YYY", "line 1:1: Column 'Y' does not exist");
+        assertUpdate("ALTER TABLE Test_Alter RENAME COLUMN y to YYY");
+
+        assertQueryFails("ALTER TABLE IF EXISTS Test_Alter DROP COLUMN notExistColumn", ".*Column 'notExistColumn' does not exist.*");
+        assertUpdate("ALTER TABLE Test_Alter DROP COLUMN YYY");
+
+        getQueryRunner().execute(session, "DROP TABLE IF EXISTS Test_Alter");
+    }
+
+    @Test
+    public void testInsert()
+    {
+        getQueryRunner().execute(session, "CREATE TABLE TEST_INSERT(name VARCHAR(50), rollNum int)");
+        assertTrue(getQueryRunner().tableExists(session, "TEST_INSERT"));
+
+        assertQueryFails("INSERT INTO Test_Insert VALUES (123, 'test')", ".*Table clickhouse.tpch.Test_Insert does not exist.*");
+
+        getQueryRunner().execute(session, "INSERT INTO TEST_INSERT VALUES ('test', 123)");
+
+        getQueryRunner().execute(session, "DROP TABLE IF EXISTS TEST_INSERT");
+    }
+}
```

Comment:
Lets use `try-finally` block for `DROP` to make sure the cleanup across all the tests